core/client/fs/src/main/java/alluxio/client/block/stream/GrpcDataMessageBlockingStream.java

```diff
@@ -79,9 +84,25 @@ public ResT receive(long timeoutMs) throws IOException {
    * @throws IOException if any error occurs
    */
   public DataMessage<ResT, DataBuffer> receiveDataMessage(long timeoutMs) throws IOException {
+    return retryReceiveDataMessage(timeoutMs, null);
+  }
+
+  /**
+   * Receives a response with data buffer from the server. Will wait until a response is received,
+   * or throw an exception if times out. Caller of this method must release the buffer after reading
+   * the data.
+   *
+   * @param timeoutMs maximum time to wait before giving up and throwing
+   *                  a {@link DeadlineExceededException}
+   * @param request   the request
+   * @return the response message with data buffer, or null if the inbound stream is completed
+   * @throws IOException if any error occurs
+   */
+  public DataMessage<ResT, DataBuffer> retryReceiveDataMessage(long timeoutMs, ReqT request)
```

Comment:
similar comment about whether a separate `retryReceive` is necessary if `receive` directly calls it.